### PR TITLE
[feature-wip](dlf) prepare to support aliyun dlf

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #xxx

## Problem summary

[What is DLF](https://www.alibabacloud.com/product/datalake-formation)

This PR is a preparation for support DLF, with some changes of multi catalog

1. Add RuntimeException for most of hive meta store or es client visit operation.
2. Add DLF related dependencies.
3. Move the checks of es catalog properties to the analysis phase of creating es catalog

TODO(in next PR):

1. Refactor the `getSplit` method to support not only hdfs, but s3-compatible object storage.
2. Finish the implementation of supporting DLF

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [x] No
    - [ ] I don't know
5. Has unit tests been added:
    - [ ] Yes
    - [x] No
    - [ ] No Need
6. Has document been added or modified:
    - [ ] Yes
    - [x] No
    - [ ] No Need
7. Does it need to update dependencies:
    - [x] Yes
    - [ ] No
8. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [x] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java [hunk 1]
```diff
@@ -135,28 +140,46 @@ public static String getFormat(String input) throws DdlException {
...
-        HiveMetaStoreClient hivemetastoreclient = null;
+        return getClient(hiveConf);
+    }
+
+    public static IMetaStoreClient getClient(HiveConf hiveConf) throws DdlException {
+        hiveConf.set(ConfVars.METASTORE_CLIENT_SOCKET_TIMEOUT.name(),
+                String.valueOf(Config.hive_metastore_client_timeout_second));
+        Properties prop = hiveConf.getAllProperties();
+        for (String key : prop.stringPropertyNames()) {
+            LOG.info("cmy debug key: {}, value: {}", key, prop.getProperty(key));
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java:154]: Set to LOG.debug?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java [hunk 2]
```diff
@@ -135,28 +140,46 @@ public static String getFormat(String input) throws DdlException {
...
+        }
+        IMetaStoreClient metaStoreClient = null;
+        String type = hiveConf.get(HIVE_METASTORE_TYPE);
+        LOG.info("cmy debug type: {}", type);
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java:158]: LOG.debug?

File: fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalCatalog.java [hunk 3]
```diff
@@ -74,61 +76,45 @@ public class EsExternalCatalog extends ExternalCatalog {
...
-        if (StringUtils.isBlank(properties.get(PROP_HOSTS))) {
-            throw new DdlException("Hosts of ES table is null.");
-        }
-        nodes = properties.get(PROP_HOSTS).trim().split(",");
-        // check protocol
-        for (String seed : nodes) {
-            if (!seed.startsWith("http")) {
-                throw new DdlException("the protocol must be used");
-            }
+    private void setProperties(Map<String, String> properties) {
```
Comment (Jibing-Li) [inline fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalCatalog.java:88]: I think we need to call `setProperties` in `gsonPostProcess` as well to make sure all properties are set after metadata replay.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.73 seconds
 load time: 447 seconds
 storage size: 17178443022 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221104123020_clickbench_pr_40075.html

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java (hunk 1 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java:154]: Removed

File: fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java (hunk 2 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/HiveMetaStoreClientHelper.java:158]: Removed

File: fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalCatalog.java (hunk 3 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/datasource/EsExternalCatalog.java:89]: Fixed

**Review by Jibing-Li — DISMISSED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**